### DISCUSSION TITLE
Custom rule with additional data not retriving ID at update form

### DISCUSSION THREAD

**Discussion by ProjectesFUAB** [laravel/nova-issues#5582] [Q&A] [answered]
When creating a custom validation rule with additional data (as indicated in the Laravel documentation) $this->data does not include the value of the field ID and neither the values of the Hidden fields.

**Comment by crynobone** [comment]
You should be able to retrieve the resourceId from `$request` object. On an update request it should be within the route parameter

**Reply by ProjectesFUAB** [reply]
Yeah,

but I'm using ->rules('required', ... , new CustomRule) as indicated in the[ Laravel documentation](https://laravel.com/docs/10.x/validation#custom-validation-rules), with 

```
class CustomRule implements DataAwareRule, ValidationRule {
   protected $data = [];

   public function setData($data) {
      $this->data = $data;
      return $this;
   }
   /**
    * Run the validation rule.
    *
    * @param  \Closure(string): \Illuminate\Translation\PotentiallyTranslatedString  $fail
    */
   public function validate(string $attribute, mixed $value, Closure $fail): void {
      ...
   }
}
```

in $this->data are all the values of the form fields except the ID and the fields of type Hidden (or using the ->hidden() method), and I need use the resource ID inside validate()

I know I can hack this behaviour, but I think that both the ID (with null value in creation forms?) and the "hidden" fields should be included in the data that the rule receives.

**Reply by crynobone** [reply, marked as answer]
No, the current behaviour is intended and you should get the ID from Request object.
